packages/react/src/components/Table/TableToolbar/TableToolbar.jsx

```diff
@@ -284,17 +310,49 @@ const TableToolbar = ({
         totalSelected={totalSelected}
         translateWithId={(...args) => tableTranslateWithId(i18n, ...args)}
       >
-        {batchActions.map(({ id, labelText, ...others }) => (
+        {visibleBatchActions.map(({ id, labelText, disabled, ...others }) => (
           <TableBatchAction
             key={id}
             onClick={() => onApplyBatchAction(id)}
             tabIndex={shouldShowBatchActions ? 0 : -1}
-            disabled={!shouldShowBatchActions}
+            disabled={!shouldShowBatchActions || disabled}
             {...others}
           >
             {labelText}
           </TableBatchAction>
         ))}
+        {hasVisibleOverflowBatchActions ? (
+          <OverflowMenu
+            data-testid={`${testID || testId}-batch-actions-overflow-menu`}
```

Comment:
do we need to add support for the deprecated testID?

Reply:
We probably should until v3, right? Then remove it with all the rest?

Reply:
I guess either way is fine, my thought was that we would not be breaking any functionality by only allowing testId for new features... if anything it would encourage developers to move away from the deprecated one.

Reply:
We could, but if they're still using `testID` and we won't add it here it'll be `undefined-batch-action...`, right?

Reply:
hehe, now it is getting complicated. It would actually use the default value for the `testId` which an empty string so we would get `-batch-action...` But the Table as only consumer of  the TableToolbar never sets a `testID` is only sets the `testId` so we could just skip the `testID` altogether ... **but** no wait, I see that we are still exporting the TableToolbar for some reason. Fun stuff :-)

Reply:
Ha! That's ridiculous. I'll remove `testID` and see what we end up with in the snapshots. Re: exporting, I _think_ we removed that in v3, just in case someone was using it for some reason.